### PR TITLE
Fix symlinks to libs on Deb10

### LINKED ISSUE #11129: Basic Auth over HTTP not allowed on Unix.New-PSSession test failing on Debian 10
Basic Auth over HTTP not allowed on Unix.New-PSSession should throw when specifying Basic Auth over HTTP on Unix

Failed 3h ago on 6dcfecda2e0b
Duration0:00:00.027
Failing buildReleaseAutomationTest-114865
OwnerPaul Higinbotham

Debug
Work items
Attachments
History

Error message
Expected 801, but got 0.

https://dev.azure.com/mscodehub/PowerShellCore/_build/results?buildId=117686&view=ms.vss-test-web.build-test-results-tab&runId=1096282&resultId=101189&paneView=debug

Stack trace
at <ScriptBlock>, /home/vsts_azpcontainer/PSPackage/download/Test/test/powershell/engine/Remoting/RemoteSession.Basic.Tests.ps1: line 32
32:         $err.Exception.ErrorCode | Should -Be 801

# Steps to reproduce

```powershell

```

# Expected behavior

```none

```

# Actual behavior

```none

```

# Environment data

```none

```

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

Importing build.psm1 fails as there's more than one lsb_release command on the system.  Fix is to choose the first one.  Also fix the version of libs required for WSMan client lib for deb10.

## PR Context

Fix https://github.com/PowerShell/PowerShell/issues/11129

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: build.psm1 [hunk 1]
```diff
@@ -134,7 +134,7 @@ function Get-EnvironmentInformation
```
Comment (PoshChan) [inline build.psm1]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-windows`
Enter-PSHostProcess tests.By Process Id.Can enter using NamedPipeConnectionInfo
```powershell
Exception calling "Invoke" with "0" argument(s): "The runspace state is not valid for this operation."
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Core\Enter-PSHostProcess.Tests.ps1: line 136
136:                 $ps.AddScript('$pid').Invoke() | Should -Be $pwshId

```

**Comment by SteveL-MSFT** [comment]
@PoshChan please retry windows

**Comment by PoshChan** [comment]
@SteveL-MSFT, successfully started retry of `PowerShell-CI-Windows`

File: build.psm1 [hunk 2]
```diff
@@ -499,13 +500,17 @@ Fix steps:
...
 
-    if ($Environment.IsRedHatFamily -or $Environment.IsDebian9) {
+    if ($Environment.IsRedHatFamily -or $Environment.IsDebian) {
         # add two symbolic links to system shared libraries that libmi.so is dependent on to handle
         # platform specific changes. This is the only set of platforms needed for this currently
         # as Ubuntu has these specific library files in the platform and macOS builds for itself
         # against the correct versions.
 
-        if ($Environment.IsDebian9){
+        if ($Environment.IsDebian10){
```
Comment (adityapatwardhan) [inline build.psm1]: We need to support Debian 11 as well since release a container image for it.

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: build.psm1 (hunk 2 above)
Comment (SteveL-MSFT) [reply build.psm1]: I'll create a Deb11 image to see if the versions of the files are same or different and will update this PR

Comment (SteveL-MSFT) [reply build.psm1]: Deb11 currently uses the same version of the libs as Deb10, so just added a check for it.

File: build.psm1 [hunk 3]
```diff
@@ -145,6 +145,8 @@ function Get-EnvironmentInformation
         $environment += @{'LinuxInfo' = $LinuxInfo}
         $environment += @{'IsDebian' = $LinuxInfo.ID -match 'debian' -or $LinuxInfo.ID -match 'kali'}
         $environment += @{'IsDebian9' = $Environment.IsDebian -and $LinuxInfo.VERSION_ID -match '9'}
+        $environment += @{'IsDebian10' = $Environment.IsDebian -and $LinuxInfo.VERSION_ID -match '10'}
+        $environment += @{'IsDebian11' = $Environment.IsDebian -and $LinuxInfo.PRETTY_NAME -match 'bullseye'}
```
Comment (SteveL-MSFT) [inline build.psm1]: Deb11 is a testing release so they don't have VERSION_ID yet

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**